### PR TITLE
{lang}[GCCcore/12.1.0] Perl v5.36.0

### PR DESCRIPTION
**Pull request by absrocks**
New Version of Perl is implemented with a GCC version of 12.1.0

### CODE REVIEW

**Comment by boegel** [comment]
@absrocks This easyconfig is using the same versions for the included extensions as `Perl-5.34.1-GCCcore-11.3.0.eb`.
Is there a particular reason for not updating those as well? (I do know that's a painful process)

**Comment by absrocks** [comment]
> @absrocks This easyconfig is using the same versions for the included extensions as `Perl-5.34.1-GCCcore-11.3.0.eb`. Is there a particular reason for not updating those as well? (I do know that's a painful process)

This easyconfig uses the latest version of zlib, expat, ncurses, DB. The libreadline new version is currently in beta version. The only latest version I found is binutils which is 2.39 but I have not tested with the new version yet.

**Comment by absrocks** [comment]
I have also added the easyconfig files of the dependencies for GCC 12.1.0.

**Comment by boegel** [comment]
@absrocks I did not mean the dependencies, but the Perl modules that are included as extensions (see `exts_list` in the easyconfig file)

**Comment by absrocks** [comment]
I have modified the config file with updated extension lists. It took a while but finally managed to add new versions of extensions.  THe new easyconfig is added in the new commit

File: easybuild/easyconfigs/p/Perl/Perl-5.36.0-GCCcore-12.1.0.eb [hunk 1]
```diff
@@ -0,0 +1,1940 @@
...
+        'checksums': ['7e6f16cc33eb1f29ff25bee51d513f4b8a84947bbfa18edb2d3cc40a2d64cafe'],
+    }),
+    ('SQL::Statement', '1.414', {
+        'source_tmpl': 'SQL-Statement-%(version)s.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/R/RE/REHSACK'],
+        'checksums': ['dde8bdcfa6a136eedda06519ba0f3efaec085c39db0df9c472dc0ec6cd781a49'],
+    }),
+    ('File::Slurp', '9999.32', {
+        'source_tmpl': 'File-Slurp-%(version)s.tar.gz',
+        'source_urls': ['https://	cpan.metacpan.org/authors/id/C/CA/CAPOEIRAB'],
```
Comment (Micket) [inline easybuild/easyconfigs/p/Perl/Perl-5.36.0-GCCcore-12.1.0.eb:1175]:
```suggestion
        'source_urls': ['https://cpan.metacpan.org/authors/id/C/CA/CAPOEIRAB'],
```

**Comment by absrocks** [comment]
@Micket, I fixed the indentation and white space error. Can you please check?

**Comment by boegel** [comment]
Looks like this is in pretty good shape. A couple of things though:

* The `ncurses` easyconfig included in this PR should be synced up with what we have in `develop` currently (see the changes in #15903). I can handle that myself quite easily to get this PR merged.
* We expect that the usage of GCC/12.1.0 as a (sub)toolchain will be minimal, since GCC/12.2.0 is already in place (and is a likely candidate as a base for the upcoming `foss/2022b` common toolchain).

@absrocks I was also wondering how you handled the update of `exts_list` in the Perl easyconfig.
Did you automate that, or is it mostly a manual job?

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
node3109.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/c0d29ab094382e726c4937833c5fa53f for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @absrocks!

**Pull request merged**